stream.c

```diff
@@ -487,21 +488,22 @@ static void* handle_md5_digest(void* param)
     }
 
     server_user = (char*)malloc((h - p->conf->stream_authentication) + 1);
-    server_pass = (char*)malloc(strlen(h) + 1);
+    server_pass_sz = strlen(h) + 1;
+    server_pass = (char*)malloc(server_pass_sz);
```

Comment:
You can remove the explicit `(char*)` cast since it's pointless here (and worse it can actually mask an undefined `malloc` error): the returned pointer from `malloc` it will be implicitly converted to `char *` already by the merit of being set to `server_pass`.

By the way, I'm aware this sort of explicit casting is happening already throughout this file, but since you're modifying this particular line, we may as well get it right here.